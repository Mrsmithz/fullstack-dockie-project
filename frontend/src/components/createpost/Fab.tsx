import React from "react";
import { IconButton, Box } from "@chakra-ui/react";
import { AddIcon } from "@chakra-ui/icons";
import Link from "next/link";

type Props = {};

const Fab = () => {
  return (
    <>
      <Box
        style={{
          position: "fixed",
          bottom: 50,
          right: 30,
        }}
      >
        <Link href="/createpost" passHref>
          <IconButton
            id="create-btn"
            borderRadius={200}
            colorScheme="blue"
            aria-label="Call Segun"
            size={"lg"}
            icon={<AddIcon />}
          />
        </Link>
      </Box>
    </>
  );
};

export default Fab;
