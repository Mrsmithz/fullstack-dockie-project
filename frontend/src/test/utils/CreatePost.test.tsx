import { render, screen } from "@testing-library/react";
import CreatePost from "../../pages/createpost/index";
import Fab from "../../components/createpost/Fab";
import "@testing-library/jest-dom";

describe("Create Post Page", () => {
  it("render create post page", () => {
    render(<CreatePost />);
    const textTitle = screen.getByText(/Create Post/i);
    expect(textTitle).toBeInTheDocument();
  });

  it("render fab", () => {
    const fab = render(<Fab/>)
    expect(fab.getByRole("button")).toHaveAttribute("href", "/createpost")
  });

});
