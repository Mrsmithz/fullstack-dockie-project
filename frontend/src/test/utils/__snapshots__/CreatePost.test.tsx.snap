// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Upload Document Page upload file 1`] = `
Array [
  <input
    autocomplete="off"
    data-testid="upload-document"
    id="upload-document"
    multiple=""
    style="display: none;"
    tabindex="-1"
    type="file"
  />,
]
`;
